src/coreclr/vm/classlayoutinfo.cpp

```diff
@@ -897,19 +897,24 @@ EEClassNativeLayoutInfo* EEClassNativeLayoutInfo::CollectNativeLayoutFieldMetada
         parentAlignmentRequirement = pParentLayoutInfo->GetLargestAlignmentRequirement();
     }
 
+    BYTE packingSize = pMT->GetLayoutInfo()->GetPackingSize();
+    if (packingSize == 0)
+    {
+        packingSize = DEFAULT_PACKING_SIZE;
+    }
+
     BYTE fieldAlignmentRequirement = 0;
     // Now compute the native size of each field
     for (LayoutRawFieldInfo* pfwalk = pInfoArray; pfwalk->m_token != mdFieldDefNil; pfwalk++)
     {
         pfwalk->m_placement.m_size = pfwalk->m_nfd.NativeSize();
-        pfwalk->m_placement.m_alignment = pfwalk->m_nfd.AlignmentRequirement();
+        pfwalk->m_placement.m_alignment = min(packingSize, (BYTE)pfwalk->m_nfd.AlignmentRequirement());
         if (pfwalk->m_placement.m_alignment > fieldAlignmentRequirement)
         {
             fieldAlignmentRequirement = (BYTE)pfwalk->m_placement.m_alignment;
         }
     }
-
-    pNativeLayoutInfo->m_alignmentRequirement = max(max<BYTE>(1, parentAlignmentRequirement), fieldAlignmentRequirement);
+    pNativeLayoutInfo->m_alignmentRequirement = max(max<BYTE>(1, min(packingSize, parentAlignmentRequirement)), fieldAlignmentRequirement);
```

Comment:
Let's decompose this statement a bit. I see two max() calls and a min() call. Can you tease this out a bit and elaborate on the intent here.